contentcuration/contentcuration/viewsets/channelset.py

```diff
@@ -50,14 +56,33 @@ class Meta:
         list_serializer_class = BulkListSerializer
 
 
+class ChannelSetFilter(FilterSet):
+    edit = BooleanFilter(method="filter_edit")
+
+    def filter_edit(self, queryset, name, value):
+        return queryset.filter(edit=True)
+
+    class Meta:
+        model = ChannelSet
+        fields = ("edit",)
+
+
 class ChannelSetViewSet(ValuesViewset):
     queryset = ChannelSet.objects.all()
     serializer_class = ChannelSetSerializer
+    filter_backends = (DjangoFilterBackend,)
+    filter_class = ChannelSetFilter
     permission_classes = [IsAuthenticated]
-    values = ("id", "name", "description", "channels", "secret_token__token")
+    values = ("id", "name", "description", "channels", "secret_token__token", "edit")
 
     field_map = {"secret_token": "secret_token__token"}
 
+    def get_queryset(self):
+        queryset = super(ChannelSetViewSet, self).get_queryset()
+        user_id = not self.request.user.is_anonymous() and self.request.user.id
+        edit = Exists(User.channel_sets.through.objects.filter(user_id=user_id, channelset_id=OuterRef("id")))
+        return queryset.annotate(edit=edit).filter(edit=True)
```

Comment:
Should remove this: `.filter(edit=True)` otherwise we are building in the filter you've added here into the base queryset.

Beyond that, this is good to go!